private/buf/bufref/bufref.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufref
+
+import (
+	"strings"
+)
+
+// GetRawPathAndOptions returns the raw path and options from the value provided,
+// the rawPath will be non-empty when returning without error here.
+func GetRawPathAndOptions(value string) (string, map[string]string, error) {
```

Comment:
Flagging @bufdev to check this one out. This stemmed from [this](https://github.com/bufbuild/buf/pull/946#discussion_r810413228) comment. Let me know if you have any other ideas.

Reply:
sgtm